postfix-3.9.yaml

```diff
@@ -0,0 +1,289 @@
+        "${{targets.contextdir}}"/usr/share/licenses/postfix/
+      mv "${{targets.contextdir}}"/etc/postfix/*.default \
+      	"${{targets.contextdir}}"/usr/share/doc/postfix/defaults/
+      mv "${{targets.contextdir}}"/etc/postfix/*LICENSE* \
+      	"${{targets.contextdir}}"/usr/share/licenses/postfix/
+      install -Dm755 postfix.initd "${{targets.contextdir}}"/etc/init.d/postfix
+      chown postfix "${{targets.contextdir}}"/var/spool/postfix/* "${{targets.contextdir}}"/var/lib/postfix
+      chown root:postfix "${{targets.contextdir}}"/var/spool/postfix/pid
+      chgrp postdrop "${{targets.contextdir}}"/var/spool/postfix/maildrop \
+      	"${{targets.contextdir}}"/var/spool/postfix/public
+
+      cd "${{targets.contextdir}}"/etc/postfix/
+      for map in ldap mysql pcre pgsql sqlite lmdb; do
+      	echo "split $map"
+      	_mvline "^\s*$map" "$map" dynamicmaps.cf
+      done
+      rm makedefs.out
+
+
+      # fix /etc/postfix/postfix-files so that "postfix set-permissions" can run
+      sed -i \
+      	-e '/shlib_directory\/postfix-/d' \
+      	-e '/meta_directory\/makedefs.out/d' \
+      	-e '/manpage_directory/d' \
+      	-e '/config_directory\/LICENSE/d' \
+      	-e '/config_directory\/TLS_LICENSE/d' \
+      	-e '/config_directory\/[^/]\+\.cf\.default/d' \
+      	"${{targets.contextdir}}"/etc/postfix/postfix-files
+
+subpackages:
+  - name: postfix-doc
+    pipeline:
+      - uses: split/manpages
+    description: postfix manpages
+
+  - name: postfix-stone
+    description: Postfix simulation and testing tools
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.contextdir}}"/usr/bin
+          install -Dm755 src/fsstone/fsstone "${{targets.contextdir}}"/usr/bin/fsstone
+          find src/smtpstone -mindepth 1 -perm 0755 -exec cp {} "${{targets.contextdir}}"/usr/bin \;
+
+  - range: _subpackages
+    name: postfix-${{range.key}}
+    description: "${{range.key}} map support for postfix"
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.contextdir}}"/usr/lib/postfix \
+                 "${{targets.contextdir}}"/etc/postfix/dynamicmaps.cf.d
+          m="${{range.key}}"
+          mkdir -p "${{targets.contextdir}}"/usr/lib/postfix \
+                 "${{targets.contextdir}}"/etc/postfix/dynamicmaps.cf.d
+          mv "${{targets.destdir}}"/usr/lib/postfix/postfix-$m.so \
+          	"${{targets.subpkgdir}}"/usr/lib/postfix/
+          mv "${{targets.destdir}}"/etc/postfix/dynamicmaps.cf.d/$m \
+          	"${{targets.subpkgdir}}"/etc/postfix/dynamicmaps.cf.d/
+
+update:
+  enabled: true
+  release-monitor:
+    identifier: 3693
+
+test:
+  pipeline:
+    - runs: |
+        #!/bin/bash
+        #create postdrop group and postfix user with unique group IDs
+
+        addgroup -S postfix
+        addgroup -S postdrop
+        adduser -S -D -H -h /dev/null -s /sbin/nologin -G postfix postfix
+        chown -R postfix:postdrop /var/spool/postfix
+        chmod -R 755 /var/spool/postfix
```

Comment:
This makes it seem like this user has to exist to run the package. Is that the case? If so, we might need to ensure the image that includes this package also has this user.

Reply:
Good catching this in the testing btw. Given this is a package request and not an image request, it's be worth digging into this a little more, as for anyone who might want the package, they'd need to be aware

Reply:
as we have this:

```
  scriptlets:
    pre-install: |
      #!/bin/sh
      addgroup -S postfix 2>/dev/null
      addgroup -S postdrop 2>/dev/null
      adduser -S -H -h /var/spool/postfix -G postfix -g postfix postfix 2>/dev/null
      addgroup postfix mail 2>/dev/null
      adduser -S -H -h /var/mail/domains -s /sbin/nologin -G postdrop -g vmail vmail 2>/dev/null
      exit 0
```

this shouldn't be any problem, am I right?